### PR TITLE
Train-test split in training and inference set in forecasting

### PR DESCRIPTION
**Pull request by mariamedp**
Main changes:
- Split original dataset by date (same as AutoML) and change references of second set to inference set (prev. test set)
- Train-test split during training. Metrics are computed on test. Extra param "test_proportion" that defaults to 0.3.
- After logging the metrics, model is retrained using the whole dataset (training part, always leaving aside inference set) and registered
- Forecasting script no longer calculates and logs metrics, it only generates predictions (as in AutoML)

### CODE REVIEW

File: Custom_Script/02_Training_Pipeline.ipynb [hunk 1]
```diff
@@ -282,15 +282,17 @@
     "               '--timestamp_column', 'WeekStarting', \n",
     "               '--timeseries_id_columns', 'Store', 'Brand',\n",
     "               '--drop_columns', 'Revenue', 'Store', 'Brand',\n",
-    "               '--model_type', 'lr'])"
+    "               '--model_type', 'lr',\n",
+    "               '--test_proportion', 0.3]\n",
```
Comment (EricWrightAtWork) [inline Custom_Script/02_Training_Pipeline.ipynb:286]: It's not a common case in forecasting to split via proportions. Splits are usually made by date or, more commonly, made so that the test/validation set contains a certain number of observations. This number usually comes from business process requirements. Would suggest changing to use one of those types of inputs

File: Custom_Script/03_Forecasting_Pipeline.ipynb [hunk 2]
```diff
@@ -105,7 +105,7 @@
    "metadata": {},
    "source": [
     "## 3.0 Get the dataset\n",
-    "Now we get a reference to the test data in our Datastore. We will use the models trained in the Modeling Notebook to generate forecasts over all rows in each test set."
+    "Now we get a reference to the inference data in our Datastore. We will use the models trained in the Modeling Notebook to generate forecasts over all rows in each file."
```
Comment (EricWrightAtWork) [inline Custom_Script/03_Forecasting_Pipeline.ipynb:108]: Personal pet peeve, possibly. But inference is not a good word for what is really prediction. It's overloaded with the statistical concept of inference which implies a pretty different use of the model.

Comment (mariamedp) [reply Custom_Script/03_Forecasting_Pipeline.ipynb:108]: The term "inference" was taken from AutoML notebooks, who were already using it, so this is out of scope for this PR, but we can of course talk about it later. I think "prediction dataset" would be a good option. But I guess "inference" was used because it goes back to the very structure of AML areas (and real-time inferencing, batch inferencing, etc are terms widely used in AML documentation)

File: Custom_Script/02_Training_Pipeline.ipynb (hunk 1 above)
Comment (mariamedp) [reply Custom_Script/02_Training_Pipeline.ipynb]: Changed to test_size to split based on number of observations.

**Review by EricWrightAtWork — APPROVED** [review]
